### PR TITLE
Adding first shot at GitHub contributing guide

### PR DESCRIPTION
**Pull request by vsoch**
This is the first shot at a skeleton for a GitHub contributing guide. There are in fact so many things to document and learn, I can't imagine this can be anything but a first shot. I invite others to write content and either PR to this branch, or just copy paste into a comment and I'll add to the PR. I tried to focus on the parts that I thought I'd be good to write (the first sections) and the actual "do this git thing" I left minimal, hoping that we can develop / write them more collaboratively. I'd even say at the bottom we should add some nice links to other "Getting started with GitHub guides" for the interested user, because we just can't cover it all, but can cover a sourcecred branded introduction.

## How should I review this?

If you are experienced with GitHub, you likely need to try to put yourselves in the shoes of a new person. It's all new, confusing, and different. If you are inexperienced, then you are the perfect person to review, because you can focus on:

 - what areas need more detail?
 - what is confusing?

Generally, I'm interested to know what details should be added (versus what is too much detail) and how we can further customize this for sourcecred. Also, what do you think about the example / suggestion to walk the new user through adding their GitHub alias to the AUTHORS.md file in this repository (which I am adding for this PR, others will need to obviously update it!) The purpose of AUTHORS.md isn't to be about formal authors, but rather to give new folks an easy way to contribute and then see an outcome. If a more substantial project is created in the future (e.g., sourcecred/getting-started or something like that) we can modify the example here.

## What is the tone here?

You'll notice that the tone is different from the previous code of conduct, and that's intentional. The tone I'm aiming for is respectful, but more fun, akin to if two people were chatting casually. I'd much rather see a new contributor reading and chuckle at a terrible joke than reading and get bored and... stop reading :)

## Relevant Links

 - [Git Workflow](https://blog.soshace.com/understanding-the-git-workflow/) or one derivation, published recently. We should probably distinguish between "a simple example command using git" (here) and "a larger strategy that a project uses" (that has slight variation) while using git.

Please feel free to edit and add links here, or comment on how we might integrate.

Signed-off-by: Vanessa Sochat <vsochat@stanford.edu>

### CODE REVIEW

File: guides/GITHUB.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+## Why do we use GitHub?
+
+For now, think of GitHub as a place to collaborate. 
+If you had a bunch of documents that you wanted to collaborate on, what would you do?
+You might consider putting them in a shared file service (Google Drive, Dropbox) 
+or perhaps keeping them as Word documents on your local machine. But what if someone
+else wanted to make changes, or ask a question? What if you wanted the most recent version
+of those documents to render in a web interface? What if you realized
+that you make a mistake, and wanted to go back to a previous version?
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: okay I'll see what I can do to re-organize and write the chunks of this section.

File: guides/GITHUB.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+ - [sourcecred-action](https://github.com/sourcecred/sourcecred-action): is a set of workflow that can be used by other GitHub communities.
+ - [sourcecred-python](https://github.com/sourcecred/sourcecred-python): is an (under development) Python client for sourcecred.
+
+If you are familiar with filesystems and permissions, you can think of each
+repository as a folder that carries it's own permissions and metadata. And each
+carries it's own kind of expertise. For example, if you are a data scientist
+and comfortable with Python, you'd be interested to contribute to `sourcecred-python`.
+If you like working with configuration files and workflows, `sourcecred-action`
+might be for you. If you aren't a programmer, you might be most comfortable starting
+with `docs`.
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: okay I'll use this metaphor.

File: guides/GITHUB.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+### How do I contribute if I'm not a programmer?
+
+It's easy to think:
+
+> I'm not a programmer, why would anyone want my help?
+
+But here's the thing (cue Monk)! When you look deeper at a lot of discussions
+that are had around GitHub repositories, while some of them might require code 
+expertise, a large portion do not. Even when the changes are specifically to
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: okay splitting up.

File: guides/GITHUB.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+### It's okay to ask for help
+
+The SourceCred community has several ways to communicate, namely Discord
+and Discourse, and if you need help... ask for it! We commonly don't
+ask for help because we are afraid of rejection, or looking
+like you we know things. But this is ironic, because someone might actually
+like you more if you ask (reference study) and being able to give help
+is hugely fulfilling for the other party. You might think the person
+doesn't get much out of the interaction, but they do. Don't forget that.
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: okay going to update this section too.

**Comment by Beanow** [comment]
Thanks :smile: the updates look good to me.
I'll wait for what others have to say from here.

File: guides/GITHUB.md [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+
+What does it mean to ask for help? You might have questions, feedback, or
+a general comment, or just want to talk to someone. Asking for help can be daunting, 
+but guess what? The SourceCred community is open to it. Once you've
+sprinted over the mental hurdle that it's okay to ask, you'll very likely
+realize that engaging in the first place is a great way to learn, and quite a bit
+of fun. 
+
+
+### I will grow
```
Comment (decentralion) [inline guides/GITHUB.md]: I'm skeptical that we should have these mindset sections at the front of the doc. It feels like, _too_ hand-holdey, personally I would rather get to content. however, i recommend I'm not the target audience for the doc -- curious for feedback from e.g. LB

File: guides/GITHUB.md [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+
+This is the fun part! If you haven't yet, you get to [create an account](https://github.com/)
+which usually comes down to generating a username that you like and providing an email.
+This is exciting because when you think about it, GitHub is just another super
+geeky social network where you'll make a lot of new friends, and connect
+with friends that you already might have on their GitHub accounts.
+Once you have an account, don't forget to upload a profile picture and add
+any other metadata of interest under [settings](https://github.com/settings/profile).
+
+### 2. Install git
```
Comment (decentralion) [inline guides/GITHUB.md]: I think we should guide folks to [GitHub for Desktop](https://desktop.github.com/) instead of using the command line. For people who are not exposed to using CLIs, they're very intimidating and a big learning curve that is outside of the scope of this gude.

**Comment by decentralion** [comment]
Note that GitHub for Desktop has a tutorial flow that guides the user through steps like getting a text editor, creating a branch, etc.

image

File: guides/GITHUB.md [hunk 7]
```diff
@@ -0,0 +1,343 @@
+# Getting Started with GitHub
+
+What is GitHub, and how do you use it to interact with and contribute to SourceCred?
+This guide will give you an introduction to GitHub and SourceCred.
+
+## Why do we use GitHub?
+
+For now, think of GitHub as a place to collaborate. 
+If you had a bunch of documents that you wanted to collaborate on, what would you do?
+You might consider putting them in a shared file service (Google Drive, Dropbox)
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: @decentralion the "playground" is a great idea, and is exactly what I was suggesting in the description (but I like your name better for it):

>  If a more substantial project is created in the future (e.g., sourcecred/getting-started or something like that) we can modify the example here.

I will remove the comparators other than Google Docs. For GitHub Desktop, I think someone else might need to write these sections (or we just link there?) I've never used it, and likely won't install it (I tend to stay away from IDEs).

File: guides/GITHUB.md [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+### More detailed history
+
+You are going to have basic questions like:
+
+ - Who added this piece?
+ - When and why was this added?
+
+And GitHub has built in tools to do this (one such action is called [git blame](https://help.github.com/en/github/managing-files-in-a-repository/tracking-changes-in-a-file).)
+
+### Different conflict resolution
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: I like this better too, changing locally.

File: guides/GITHUB.md [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+Yes! Any substantial software project is multi-disciplinary. This one included. In fact, calling SourceCred a software project doesn't do justice to how much work besides programming goes into it.
+
+
+### Why and how should I get involved here specifically?
+
+It's easy to think:
+
+> GitHub seems like a platform for programmers. I'm not a programmer, why would anyone want my help?
+
+But here's the thing (cue Monk)! When you look deeper at a lot of discussions
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: It's a reference to [this Monk](https://en.wikipedia.org/wiki/Monk_(TV_series)), which is my current favorite thing :) Now I'm just embarrassed, I'll remove it.

File: guides/GITHUB.md [hunk 10]
```diff
@@ -0,0 +1,343 @@
...
+are capable of, of your innate abilities.
+
+### It's okay to ask for help
+
+What does it mean to ask for help? You might have questions, feedback, or
+a general comment, or just want to talk to someone. Asking for help can be daunting, 
+but guess what? The SourceCred community is open to it. Once you've
+sprinted over the mental hurdle that it's okay to ask, you'll very likely
+realize that engaging in the first place is a great way to learn, and quite a bit
+of fun.
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: Added.

File: guides/GITHUB.md (hunk 6 above)
Comment (vsoch) [reply guides/GITHUB.md]: okay I modified it to remove the command line examples an direct folks either there (or more advanced) the command line getting started. Some new folks might be interested in at least reading how it goes on the command line.

File: guides/GITHUB.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: okay I did add GitHub Desktop, and it was relatively easy to link to their getting started guide.

File: guides/GITHUB.md [hunk 11]
```diff
@@ -34,7 +33,7 @@ You are going to have basic questions like:
 
 And GitHub has built in tools to do this (one such action is called [git blame](https://help.github.com/en/github/managing-files-in-a-repository/tracking-changes-in-a-file).)
```
Comment (s-ben) [inline guides/GITHUB.md]: built in -> built-in

Comment (vsoch) [reply guides/GITHUB.md]: Thanks!

File: guides/GITHUB.md [hunk 12]
```diff
@@ -0,0 +1,308 @@
...
+ 2. keep a record of discussions, and every change made
+ 3. be able to revert back to previous versions, if needed
+ 4. render some of the static files in a web interface
+
+## What content do we have on GitHub?
+
+Generally, anything that you'd want to put in a digital safe deposit box is what you'll
+find on GitHub. This obviously includes code, but also includes documentation and
+other web content. Specifically, the [sourcecred organization](https://github.com/sourcecred)
+has the follow sets of files. Each set is called a "repository" because it has it's own link
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: Fixed!

File: guides/GITHUB.md [hunk 13]
```diff
@@ -0,0 +1,308 @@
...
+and space on the GitHub website.
+
+ - [sourcecred](https://github.com/sourcecred/sourcecred): is the core sourcecred codebase.
+ - [docs](https://github.com/sourcecred/docs): is a growing set of documentation and community health files. In fact, the file that you are reading right now is part of it!
+ - [sourcecred-action](https://github.com/sourcecred/sourcecred-action): is a set of workflow that can be used by other GitHub communities.
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: Also fixed.

File: guides/GITHUB.md [hunk 14]
```diff
@@ -0,0 +1,308 @@
...
+ - [sourcecred-python](https://github.com/sourcecred/sourcecred-python): is an (under development) Python client for sourcecred.
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: I decided to remove it all-together, there's really nothing to look at there anyhoo, and I think this is what others in the community would suggest.

File: guides/GITHUB.md [hunk 15]
```diff
@@ -0,0 +1,308 @@
...
+
+## What content do we have on GitHub?
+
+Generally, anything that you'd want to put in a digital safe deposit box is what you'll
+find on GitHub. This obviously includes code, but also includes documentation and
+other web content. Specifically, the [sourcecred organization](https://github.com/sourcecred)
+has the follow sets of files. Each set is called a "repository" because it has it's own link
+and space on the GitHub website.
+
+ - [sourcecred](https://github.com/sourcecred/sourcecred): is the core sourcecred codebase.
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md]: Remove instances of "is."

**Comment by vsoch** [comment]
haha, just stumbled on an "alternative starting guide." :) 

https://xkcd.com/1597/
image

File: guides/GITHUB.md [hunk 16]
```diff
@@ -0,0 +1,334 @@
...
+### Permissionless collaboration
+
+Anyone can propose changes. You don't need to be given permission to view or edit
+a document, as is required by some of the platforms mentioned previously.
+
+### More advanced reviewing tools
+
+When you open up a Google Doc, you might be able to comment or write in revision mode, but
+when you do so, you are working with the entire document. GitHub allows
+review on the level of change. Along with this more detailed ability to edit,
```
Comment (Bex2594) [inline guides/GITHUB.md:23]: Not sure what 'GitHub allows review on the level of change' means.

File: guides/GITHUB.md [hunk 17]
```diff
@@ -0,0 +1,334 @@
...
+
+ - Who added this piece?
+ - When and why was this added?
+
+And GitHub has built-in tools to do this (one such action is called [git blame](https://help.github.com/en/github/managing-files-in-a-repository/tracking-changes-in-a-file).)
+
+### Easy for many people to collaborate at once
+
+You can work in your own "alternate reality" without interfering with other work, even simultaneously. When ready these alternatives can be merged back together.
+You can also work on changes out of order, although this is more of an advanced topic.
```
Comment (Bex2594) [inline guides/GITHUB.md:39]: Why is there an order? I'm not sure what kind of order there would be for editing, other than top to bottom. A bit unclear to a non-tech person like me :)

File: guides/GITHUB.md [hunk 18]
```diff
@@ -0,0 +1,334 @@
...
+
+### Web Interfaces
+
+By way of some helpful features, the text files that you are working on can be served as web pages
```
Comment (Bex2594) [inline guides/GITHUB.md:43]: What does 'can be served as web pages' mean?

File: guides/GITHUB.md [hunk 19]
```diff
@@ -0,0 +1,334 @@
...
+directly from where they live. You can read about this feature, called [GitHub Pages](https://pages.github.com/).
+
+To summarize and answer this question, we use GitHub because we want to:
+
+ 1. easily collaborate on common files
+ 2. have advanced reviewing tools and version history
+ 3. keep a record of discussions, and every change made
+ 4. be able to revert back to previous versions, if needed
+ 5. render some of the static files in a web interface
```
Comment (Bex2594) [inline guides/GITHUB.md:52]: What do render, static files, and web interface mean?

File: guides/GITHUB.md [hunk 20]
```diff
@@ -0,0 +1,334 @@
...
+
+Generally, anything that you'd want to put in a digital safe deposit box is what you'll
+find on GitHub. This obviously includes code, but also includes documentation and
+other web content. Specifically, the [sourcecred organization](https://github.com/sourcecred)
+has the following sets of files. Each set is called a "repository" and you can think of
+it like a (sub)-project under a GitHub organization, or a GitHub username. Each repository
+has it's own link and space on the GitHub website.
+
+ - [sourcecred](https://github.com/sourcecred/sourcecred): the core sourcecred codebase.
+ - [docs](https://github.com/sourcecred/docs): is a growing set of documentation and community health files. In fact, the file that you are reading right now is part of it!
```
Comment (Bex2594) [inline guides/GITHUB.md:65]: I'm not sure 'community health files' is how I'd describe the docs file. A) I'm unsure what that means and B) I think it will be changing relatively soon, so maybe someone can incorporate a description of what the #new docs folder will include. @decentralion

File: guides/GITHUB.md [hunk 21]
```diff
@@ -0,0 +1,334 @@
...
+ - [sourcecred-action](https://github.com/sourcecred/sourcecred-action): a set of workflow that can be used by other GitHub communities.
+
+If you are familiar with filesystems and permissions, you can think of each
+repository as a folder or space that carries it's own permissions, labels,
```
Comment (Bex2594) [inline guides/GITHUB.md:69]: its* own permissions (no apostrophe)

File: guides/GITHUB.md [hunk 22]
```diff
@@ -0,0 +1,334 @@
...
+and sets of issues, pull requests (requests for changes), milestones, and sometimes
+even guides for contribution. Each is an independent entity under the organization, and thus each
+carries it's own kind of expertise. For example,
```
Comment (Bex2594) [inline guides/GITHUB.md:72]: its* own kind of expertise (no apostrophe)

File: guides/GITHUB.md [hunk 23]
```diff
@@ -0,0 +1,334 @@
...
+even guides for contribution. Each is an independent entity under the organization, and thus each
+carries it's own kind of expertise. For example, 
+if you like working with configuration files and workflows, `sourcecred-action`
+might be for you. If you aren't a programmer, you might be most comfortable starting
+with `docs`.
+
+
+### Is my help needed in a software project when I'm not a programmer?
+
+Yes! Any substantial software project is multi-disciplinary. This one included. In fact, calling SourceCred a software project doesn't do justice to how much work besides programming goes into it.
```
Comment (Bex2594) [inline guides/GITHUB.md:80]: I super appreciate this insight. It makes me feel super valued and capable of contributing. So thank you :)

File: guides/GITHUB.md [hunk 24]
```diff
@@ -0,0 +1,334 @@
...
+
+To summarize, if you have a growth mindset, and aren't afraid to ask for help,
+the experience can not only be successful in that you will learn how to use
+GitHub, you'll also have fun and connect with others. 
+
+
+## How do I contribute?
+
+Okay, let's get down to talking about GitHub! We could start by giving you high
+level concepts first, but intsead we are going to walk you through steps
```
Comment (Bex2594) [inline guides/GITHUB.md:152]: instead* (typo)

File: guides/GITHUB.md [hunk 25]
```diff
@@ -0,0 +1,334 @@
...
+> GitHub seems like a platform for programmers. I'm not a programmer, why would anyone want my help?
+
+But here's the thing! When you look deeper at a lot of discussions
+that are had around GitHub repositories, while some of them might require code 
+expertise, a large portion do not. Even when the changes are specifically to
+suggest changes to code, it's usually the responsibility of the individual
+asking for feedback to explain their changes in human friendly terms. For
+example, a developer might say something like:
+
+> This (review request) is going to add a plugin to support reddit as a place to get cred.
```
Comment (Bex2594) [inline guides/GITHUB.md:96]: To me, the purpose of this part is to say, "Here's some tech jargon that might be said that you might not understand, and here's how you could contribute regardless of not understanding it." I think that's great. However, I think the action points after the jargon could be clearer. As a beginner, I would read the jargon, not get it, and then not understand the action points presented. The questions beginners would be asking would probably be a bit less complex.

File: guides/GITHUB.md [hunk 26]
```diff
@@ -0,0 +1,334 @@
...
+
+Even if you know or understand nothing about the code, you can contribute by
+asking questions. What are the actions that can generate it? How has this
```
Comment (Bex2594) [inline guides/GITHUB.md:99]: generate it, does 'it' mean cred?

File: guides/GITHUB.md [hunk 27]
```diff
@@ -0,0 +1,334 @@
...
+
+This is the fun part! If you haven't yet, you get to [create an account](https://github.com/)
+which usually comes down to generating a username that you like and providing an email.
+This is exciting because when you think about it, GitHub is just another super
+geeky social network where you'll make a lot of new friends, and connect
+with friends that you already might have on their GitHub accounts.
+Once you have an account, don't forget to upload a profile picture and add
+any other metadata of interest under [settings](https://github.com/settings/profile).
+
+### 2. Install git
```
Comment (Bex2594) [inline guides/GITHUB.md:165]: For people who don't already know how to use GitHub, there's a low chance they would know how to use a terminal (I certainly don't). I think this section could be more useful if it went into how to contribute on GitHub specifically using the GitHub website, such as how to review documents, etc. We've recently created a document going into that, and maybe this could link to or include an official guide once it's made (there's definitely some overlap, but both will be useful IMO).
I'd also go into how a new community member would know what there is that needs working on and how to get started. @LexBirdie would know which directions to point folx in.

File: guides/GITHUB.md [hunk 28]
```diff
@@ -0,0 +1,334 @@
+# Getting Started with GitHub
+
+What is GitHub, and how do you use it to interact with and contribute to SourceCred?
+This guide will give you an introduction to GitHub and SourceCred.
```
[the remark this answers is not in the document]
Comment (vsoch) [reply guides/GITHUB.md:4]: Sure thing.

File: guides/GITHUB.md (hunk 16 above)
Comment (vsoch) [reply guides/GITHUB.md:23]: I think I was trying to get at "You can check out a new branch, and instead of globally editing the whole thing, work on your change separately from the individual document." It's definitely not clear! Do you have a suggestion for how it could be said?

Comment (vsoch) [reply guides/GITHUB.md:23]: Maybe something like:
> When you open up a Google Doc, you might be able to comment or write in revision mode, but
when you do so, you are working with the entire document. GitHub allows
you to separate your working changes from the production version, ensuring that review and discussion
can proceed before anything is made final. By way of branches, several people can do this at once
without it being confusing. Along with this more detailed ability to edit,
there is an integrated approval workflow, straightforward visualizations of what changed 
(these are called "diffs"), and powerful automatic testing (called continuous integration or CI).
 
?

File: guides/GITHUB.md (hunk 17 above)
Comment (vsoch) [reply guides/GITHUB.md:39]: You're right - there isn't any order ingrained, I was thinking of how in some cases developers have a sense of when they want changes to go in, but that's probably overkill for this introduction.

Comment (vsoch) [reply guides/GITHUB.md:39]: For an example, imagine you have a library with a bunch of PRs coming in for new features intended for the next version. You also might have PRs coming in for bug fixes for the current. Both of those sets of work can go on at the same time, or even the "for the next version" work could come before the bug fix work. This is what I meant by out of order, you don't absolutely have to do what chronologically needs to be merged into master first.

Comment (vsoch) [reply guides/GITHUB.md:39]: (Note that I'm just removing this line).

File: guides/GITHUB.md (hunk 18 above)
Comment (vsoch) [reply guides/GITHUB.md:43]: It means if you have content that can be rendered on a static web server (generally a *.html file) you can deploy it to GitHub pages (essentially choosing a branch/subfolder on your repo) and it will render as a website.

Comment (vsoch) [reply guides/GITHUB.md:43]: How would you like this stated to be more clear?

File: guides/GITHUB.md (hunk 19 above)
Comment (vsoch) [reply guides/GITHUB.md:52]: It's the previous point above! Let me know how I can change it to make it more clear.

Comment (vsoch) [reply guides/GITHUB.md:52]: A little background on the [render terminology](https://www.freecodecamp.org/news/what-exactly-is-client-side-rendering-and-hows-it-different-from-server-side-rendering-bd5c786b340d/). Static files mean that they are just raw text, not a binary that needs to be executed, and web interface is just your browser (the little window you are looking at now!)

File: guides/GITHUB.md (hunk 20 above)
Comment (vsoch) [reply guides/GITHUB.md:65]: I chose this term because we (at least used to) have a "community" folder in docs. Let me know what you would like that referred to instead.

Comment (vsoch) [reply guides/GITHUB.md:65]: It's also a GitHub known idea: https://github.blog/changelog/2019-02-21-organization-wide-community-health-files/

File: guides/GITHUB.md (hunk 21 above)
Comment (vsoch) [reply guides/GITHUB.md:69]: Looks like I had another below that too! Oups :)

File: guides/GITHUB.md (hunk 22 above)
Comment (vsoch) [reply guides/GITHUB.md:72]: Haha, totally just saw this one too :)

File: guides/GITHUB.md (hunk 23 above)
Comment (vsoch) [reply guides/GITHUB.md:80]: Definitely! A lot of the stronger (and more fun) communities have a welcoming culture like this, because it really is true that there is a ton of different things to work on. If a community is entirely, say, old school ornery developers, it's going to have a harder time promoting it's message or software to any other kind of person. I've seen it again and again - openness in communication, diversity, and making things fun works really well. As soon as something isn't fun? Why would anyone want to stick around, unless they absolutely had to.

File: guides/GITHUB.md (hunk 24 above)
Comment (vsoch) [reply guides/GITHUB.md:152]: Fixed locally.

File: guides/GITHUB.md (hunk 25 above)
Comment (vsoch) [reply guides/GITHUB.md:96]: I cleaned this up locally - will push with this round of changes after we discuss the open ended points!

File: guides/GITHUB.md (hunk 26 above)
Comment (vsoch) [reply guides/GITHUB.md:99]: Fixed this locally.

File: guides/GITHUB.md (hunk 27 above)
Comment (vsoch) [reply guides/GITHUB.md:165]: Hmm, what about if we have this be the end of the document, and then link to "Your first look at the GitHub website" (what you are referring to) and "Using git from a command line" (second, what is currently here). What I can do for now is move this last chunk of content into a separate file (something like github-terminal.md ?) and then link from here, and then you can add the link to the other document when it's ready. Does that work? I do think both descriptions are useful - a new person might be interested to try a command line (I was when I first got an Ubuntu laptop early in graduate school, although I wasn't great doing things in a terminal because I came from Windows!)

File: guides/GITHUB.md (hunk 18 above)
Comment (Bex2594) [reply guides/GITHUB.md:43]: I'm not sure I understand your second description. Are you saying that what you're working on (on Github) can be turned into its own website? Or that you can preview what your work would look like on the webpage you're a part of?

Comment (vsoch) [reply guides/GITHUB.md:43]: GitHub pages ->https://pages.github.com/ let's you turn a repository, e.g. here is one of mine -> https://github.com/vsoch/vsoch.github.io into a rendered website, e.g., here is the rendering of those files. https://vsoch.github.io. That is what I am referring to when I say serving web pages. The repository is serving static files to render on the web server that is provided by GitHub.

Comment (vsoch) [reply guides/GITHUB.md:43]: The same is true for the docs repository here...

File: guides/GITHUB.md (hunk 19 above)
Comment (Bex2594) [reply guides/GITHUB.md:52]: I ~think~ I get it... Explaining each term with simple examples might be the move here for non-tech peeps. Raw text could mean markdown, or just 'writing', like on a Word doc. When you say binary all I know is that that's all 0s and 1s, but not what that means for a website. My ask would be for an explanation of all the terms using simple vocabulary and accessible examples.

File: guides/GITHUB.md (hunk 23 above)
Comment (Bex2594) [reply guides/GITHUB.md:80]: Yes. Yes. Andddd yes. I'd be curious to learn which communities are like this to see how successful diverse communities are.

File: guides/GITHUB.md (hunk 27 above)
Comment (Bex2594) [reply guides/GITHUB.md]: That's a good point, I definitely wouldn't want to discourage anyone from learning how to use a command line. If there's a description of what knowledge you need before diving into the next link, that would totally work!

File: guides/GITHUB.md (hunk 18 above)
Comment (vsoch) [reply guides/GITHUB.md:43]: Updated with example.

File: guides/GITHUB.md (hunk 19 above)
Comment (vsoch) [reply guides/GITHUB.md]: Updated.

File: guides/GITHUB.md (hunk 23 above)
Comment (vsoch) [reply guides/GITHUB.md:81]: Great!

File: guides/GITHUB.md (hunk 16 above)
Comment (Bex2594) [reply guides/GITHUB.md]: This is perfect. Thanks!

File: guides/GITHUB.md (hunk 27 above)
Comment (vsoch) [reply guides/GITHUB.md]: I think most people are familiar enough with the concept of a command line to know if they would want to read that tutorial, so I'm not sure how you would want me to change that. If you have a suggestion for what to say I'd be happy to adjust.

File: guides/GITHUB.md (hunk 16 above)
Comment (Bex2594) [reply guides/GITHUB.md]: This is perfect! Thanks!

File: guides/GITHUB.md (hunk 20 above)
Comment (Bex2594) [reply guides/GITHUB.md:66]: Interesting. Then let's just keep it as is and if anyone disagrees they can speak up later.

File: guides/GITHUB.md (hunk 27 above)
Comment (vsoch) [reply guides/GITHUB.md]: To play devil's advocate, if we provide too much description about things that a person "needs to know" in order to read a guide, that's actually creating more reasons for them to think "Oh, that's beyond what I know / am capable of doing." I think it's a much better outcome to have someone click on a guide to get more information (and then possibly try things out or learn something) than put it right up front and scare them off the bat. But that's just me. :)

**Comment by decentralion** [comment]
I'm closing this because we're declaring bankruptcy on old docs as we streamline and prep for the beta launch.
